src/main/java/net/fabricmc/loom/configuration/CompileConfiguration.java

```diff
@@ -157,8 +157,16 @@ private synchronized void setupMinecraft(ConfigContext configContext) throws Exc
 
 		var jarConfiguration = extension.getMinecraftJarConfiguration().get();
 
-		if (jarConfiguration == MinecraftJarConfiguration.MERGED && !metadataProvider.getVersionMeta().isVersionOrNewer(Constants.RELEASE_TIME_1_3)) {
-			jarConfiguration = MinecraftJarConfiguration.LEGACY_MERGED;
+		if (jarConfiguration == MinecraftJarConfiguration.MERGED) {
+			// if no configuration is selected by the user, attempt to select one
+			// based on the mc version and which sides are present for it
+			if (!metadataProvider.getVersionMeta().downloads().containsKey("server")) {
+				jarConfiguration = MinecraftJarConfiguration.CLIENT_ONLY;
```

Comment:
No, this is wrong. It should set the value in the extension, otherwise all the other places like use this configuration will not be in agreement.

Reply:
It would likely be best to move all of this logic into the convension as a Provider. I do worry that the metadata might not have been populated if someone requests the value too soon though.

Reply:
currently the MetadataProvider is only available through the MinecraftProvider, it'd require adding that to the extension to use it in the configuration provider

Reply:
It already is part of the internal extenstion, you have the project, so you can use `LoomGradleExtension.get` to get the internal one.

Reply:
Alright I moved it into the convention. It throws an error if the metadata provider is accessed to soon as well.